L1Trigger/L1TCalorimeter/plugins/L1TStage2CaloLayer2Comp.cc

```diff
@@ -0,0 +1,1105 @@
+    dataIt = taus->begin();
+  }
+
+  // check length of collections
+  if (taus->size() != emulCol->size(currBx)) {
+    edm::LogProblem("l1tcalol2ebec")
+      << "Tau collection size difference: "
+      << "data = " << taus->size()
+      << " emul = " << emulCol->size(currBx)
+      << std::endl;
+    return false;
+  }
+
+  // processing continues only of length of data collections is the same
+  if (dataIt != taus->end() ||
+      emulBxIt != emulCol->end(currBx)) {
+
+    int nPos=0;
+    int nNeg=0;
+
+    while(true) {
+
+      bool posGood = true;
+      bool etGood = true;
+
+      // object Et mismatch
+      if (dataIt->hwPt() != emulBxIt->hwPt()) {
+	etGood = false;
+	eventGood = false;
+      }
+
+      // object position mismatch (phi)
+      if (dataIt->hwPhi() != emulBxIt->hwPhi()) {
+	posGood = false;
+      }
+
+      // object position mismatch (eta)
+      if (dataIt->hwEta() != emulBxIt->hwEta()) {
+	posGood = false;
+      }
+
+      //bypass sorting bug
+      if(etGood && !posGood){
+	l1t::TauBxCollection::const_iterator emulItCheckSort;
+	l1t::TauBxCollection::const_iterator dataItCheckSort;
+	for(emulItCheckSort = emulCol->begin(currBx),
+	      dataItCheckSort = taus->begin();
+	    emulItCheckSort != emulCol->end(currBx),
+	      dataItCheckSort != taus->end();
+	    ++emulItCheckSort, ++dataItCheckSort){
+
+	  if(dataItCheckSort->hwEta() > 0) ++nPos;
+	  if(dataItCheckSort->hwEta() < 0) ++nNeg;
+
+	  if(dataIt->hwPt() == emulItCheckSort->hwPt()
+	     && dataIt->hwPhi() == emulItCheckSort->hwPhi()
+	     && dataIt->hwEta() == emulItCheckSort->hwEta())
+	    posGood = true;
+	}
+      }
+
+      if(etGood && dataIt->hwEta() > 0 && ((distance(dataIt, taus->end())-nNeg) < 5))
+	posGood = true;
+      if(etGood && dataIt->hwEta() < 0 && (distance(dataIt, taus->end()) < 5))
+	posGood = true;
+
+      if(!posGood) eventGood = false;
+
+      // if both position and energy agree, object is good
+      if (!posGood || !etGood) {
+	edm::LogProblem("l1tcalol2ebec")
+	  << "Tau Problem (data emul): "
+	  << "\tEt = " << dataIt->hwPt() << " " << emulBxIt->hwPt()
+	  << "\teta = " << dataIt->hwEta() << " " << emulBxIt->hwEta()
+	  << "\tphi = " << dataIt->hwPhi() << " " << emulBxIt->hwPhi()
+	  << std::endl;
+      }
+
+      // increment position of pointers
+      ++dataIt;
+      ++emulBxIt;
+
+      if (dataIt == taus->end() ||
+	  emulBxIt == emulCol->end(currBx))
+	break;
+    }
+  } else {
+    if (!taus->empty() || emulCol->size(currBx) != 0)
+      return false;
+  }
+
+  // return a boolean that states whether the tau data in the event are in
+  // agreement
+  return eventGood;
+}
+
+// comparison method for sums
+bool L1TStage2CaloLayer2Comp::compareSums(
+  const edm::Handle<l1t::EtSumBxCollection> & dataCol,
+  const edm::Handle<l1t::EtSumBxCollection> & emulCol)
+{
+  bool eventGood = true;
+
+  double dataEt = 0;
+  double emulEt = 0;
+
+  l1t::EtSumBxCollection::const_iterator dataIt = dataCol->begin(currBx);
```

Comment:
This two vars are still present because this PR is missing https://github.com/cms-sw/cmssw/pull/24242/commits/d0184bacfa6874be3cf28b04f6e1bacdfb4c47b8 from https://github.com/cms-sw/cmssw/pull/24242